packages/common-integration-tests/src/item-type.ts

```diff
@@ -0,0 +1,187 @@
+import {v4} from 'uuid';
+import {
+  ValidateCacheProps,
+  ItBehavesLikeItValidatesCacheName,
+  expectWithMessage,
+} from './common-int-test-utils';
+import {ICacheClient} from '@gomomento/sdk-core/dist/src/internal/clients/cache';
+import {ItemType} from '@gomomento/sdk-core';
+import {cache_client} from '@gomomento/generated-types/dist/cacheclient';
+import _ItemGetTypeResponse = cache_client._ItemGetTypeResponse;
+
+export function runItemTypeTest(
+  Momento: ICacheClient,
+  IntegrationTestCacheName: string
+) {
+  describe('item type', () => {
+    ItBehavesLikeItValidatesCacheName((props: ValidateCacheProps) => {
+      return Momento.itemType(props.cacheName, v4());
+    });
+
+    it('should get item type scalar', async () => {
+      const cacheKey = v4();
+      const cacheValue = v4();
+      await Momento.set(IntegrationTestCacheName, cacheKey, cacheValue);
+
+      // string cache key
+      let itemTypeResponse = await Momento.itemType(
+        IntegrationTestCacheName,
+        cacheKey
+      );
+      expectWithMessage(() => {
+        expect(itemTypeResponse).toBeInstanceOf(ItemType.Hit);
+      }, `expected HIT but got ${itemTypeResponse.toString()}`);
+      let hitResult = itemTypeResponse as ItemType.Hit;
+      expect(hitResult.getItemType()).toEqual(
+        _ItemGetTypeResponse.ItemType.SCALAR
+      );
+
+      // byte array cache key
+      itemTypeResponse = await Momento.itemType(
+        IntegrationTestCacheName,
+        new TextEncoder().encode(cacheKey)
+      );
+      expectWithMessage(() => {
+        expect(itemTypeResponse).toBeInstanceOf(ItemType.Hit);
+      }, `expected HIT but got ${itemTypeResponse.toString()}`);
+      hitResult = itemTypeResponse as ItemType.Hit;
+      expect(hitResult.getItemType()).toEqual(
+        _ItemGetTypeResponse.ItemType.SCALAR
```

Comment:
yeah, here you have leaked the generated enum from one of the two protos into the tests, we don't want this.  we will need to define our own enum in core and provide a mapping from each of the two sdks.